src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'Sort alphabetically',
+  Length = 'Sort by length',
+}
+
+type Button = {
+  name: SortType | 'Reverse',
+  color: 'is-info' | 'is-success' | 'is-warning',
+};
+
+const buttons: Button[] = [
+  {
+    name: SortType.Alphabet,
+    color: 'is-info',
+  },
+  {
+    name: SortType.Length,
+    color: 'is-success',
+  },
+  {
+    name: 'Reverse',
+    color: 'is-warning',
+  },
+];
+
+let goods = [...goodsFromServer];
+
+function operateGoods(stateSort: string | null, stateReverse: string | null) {
+  switch (stateSort) {
+    case SortType.Alphabet:
+      goods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.Length:
+      goods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+      break;
+  }
+
+  if (stateReverse) {
+    goods = goods.reverse();
+  }
+
+  return '';
+}
+
 export const App: React.FC = () => {
+  const [stateSort, setStateSort] = useState<SortType | 'Reverse' | null>(null);
+  const [stateReverse, setStateReverse] = useState<'Reverse' | null>(null);
```

Comment:
better to set just boolean type
```suggestion
  const [stateReverse, setStateReverse] = useState(false);
```